deepliif/models/networks.py

```diff
@@ -321,7 +328,7 @@ class ResnetGenerator(nn.Module):
     We adapt Torch code and idea from Justin Johnson's neural style transfer project(https://github.com/jcjohnson/fast-neural-style)
     """
 
-    def __init__(self, input_nc, output_nc, ngf=64, norm_layer=nn.BatchNorm2d, use_dropout=False, n_blocks=6, padding_type='reflect'):
+    def __init__(self, input_nc, output_nc, ngf=64, norm_layer=nn.BatchNorm2d, use_dropout=False, n_blocks=6, padding_type='zero'):
```

Comment:
We could leave here the original default and eventually pass the `padding_type` as a param of the CLI command?

```suggestion
    def __init__(self, input_nc, output_nc, ngf=64, norm_layer=nn.BatchNorm2d, use_dropout=False, n_blocks=6, padding_type='reflect'):
```

Reply:
@ricdodds I like this suggestion. It provides more flexibility and preserves the old default behavior! Can you make `padding_type` a training param? I will then pick up your change and add special handling for deterministic training (maybe warning messages emphasizing that the current padding type "reflect" does not have a deterministic implementation)

Reply:
Please add padding_type as a parameter to training (in my offline version, I have done this). The 'reflect' works better than 'zero', only in case we need to train the deterministic version of the model, we will change the padding_type to 'zero'.